.github/workflows/src/arm-auto-signoff-preview.js

```diff
@@ -0,0 +1,170 @@
+
+  if (!allLabelsMatch) {
+    return LabelAction.Remove;
+  }
+
+  const checkRuns = (
+    await github.rest.checks.listForRef({
+      owner: owner,
+      repo: repo,
+      ref: head_sha,
+    })
+  ).data.check_runs;
+
+  const swaggerLintDiffs = checkRuns.filter(
+    (run) => run.name === "Swagger LintDiff",
+  );
+
+  if (swaggerLintDiffs.length > 1) {
+    throw new Error(
+      `Unexpected number of checks named 'Swagger LintDiff': ${swaggerLintDiffs.length}`,
+    );
+  }
+
+  const swaggerLintDiff =
+    swaggerLintDiffs.length === 1 ? swaggerLintDiffs[0] : undefined;
+
+  if (swaggerLintDiff && swaggerLintDiff.status === "completed") {
+    return swaggerLintDiff.conclusion === "success"
+      ? LabelAction.Add
+      : LabelAction.Remove;
+  } else {
+    // No-op if check is missing or not completed, to prevent frequent remove/add label as checks re-run
+    return LabelAction.None;
+  }
+}
+
+/**
+ * @param {import('github-script').AsyncFunctionArguments['core']} core
+ * @returns {Promise<boolean>} True if PR contains changes to existing RPs, and no new RPs
+ */
+async function incrementalChangesToExistingResourceProvider(core) {
+  core.info("incrementalChangesToExistingResourceProvider()");
+
+  const changedSwaggerFiles = await getChangedSwaggerFiles(
+    core,
+    "HEAD^",
+    "HEAD",
+    "",
+  );
+  const changedRmFiles = changedSwaggerFiles.filter((f) =>
+    f.includes("/resource-manager/"),
+  );
+
+  core.info(
+    `Changed files containing path '/resource-manager/': ${changedRmFiles}`,
+  );
+
+  if (changedRmFiles.length == 0) {
+    core.info(
+      "No changes to swagger files containing path '/resource-manager/'",
+    );
+    return false;
+  } else {
+    for (const file of changedRmFiles) {
+      if (!(await specFolderExistsInTargetBranch(file, core))) {
+        core.info(`Appears to add a new RP: ${file}`);
+        return false;
+      }
+    }
+    core.info("Appears to change an existing RPs, but adds no new RPs");
+    return true;
+  }
+}
+
+/**
+ * @param {import('github-script').AsyncFunctionArguments['core']} core
+ * @param {string} file
+ * @returns {Promise<boolean>} True if the spec folder exists in the target branch
+ */
+async function specFolderExistsInTargetBranch(file, core) {
+  core.info(`specFolderExistsInTargetBranch("${file}")`);
+
+  // Example1: specification/contosowidgetmanager/resource-manager/Microsoft.Contoso/preview/2021-10-01-preview/contoso.json
+  // Example2: specification/contosowidgetmanager/resource-manager/Microsoft.Contoso/contosoGroup1/preview/2021-10-01-preview/contoso.json
+
+  // Example1: specification/contosowidgetmanager/resource-manager/Microsoft.Contoso
+  // Example2: specification/contosowidgetmanager/resource-manager/Microsoft.Contoso/contosoGroup1
+  const specDir = path.dirname(path.dirname(path.dirname(file)));
+  core.info(`specDir: ${specDir}`);
+
+  const resultString = await lsTree("HEAD^", specDir, core);
```

Comment:
Wouldn't it by definition be in the branch that is checked out? Or are you worried something might dirty the git clone and generate this file?

Reply:
This code checks if a given folder is in the "target" (aka "base") branch of the PR.  I think "base" is the official term so I will change my code.

So for a PR to main, this code checks if the impacted spec folder already exists in main (aka `HEAD^`), or if the PR is adding the directory for the first time.

Reply:
I think I understand the confusion now, I was using the term "target" in different places to mean opposite things!

```
export async function getChangedFiles(
  core,
  baseCommitish = "HEAD^",
  targetCommitish = "HEAD",
  diffFilter = "d",
) {
```

I will ensure all the code in this PR aligns with the GH standard and/or our existing code.

Reply:
Fixed existing code to use "base/head" consistently:

https://github.com/Azure/azure-rest-api-specs/pull/32526

Will also fix any new code in this PR to align.

Reply:
I do worry a little bit about this check when it is running locally. While there is always a merge commit in an action/pipeline HEAD^ will point to the base, that is not true if running locally. Perhaps this check is not one we can easily run locally because it depends a lot on the state of a PR but just something I want to call out.

Reply:
None of the code in this PR (or any code under `.github`) is intended for running locally.  Currently, any code intended to be run locally is under `eng/tools`, and `.github` is only for code that only runs in GH Actions.

The only supported way to run code under `.github` locally is via the unit tests (which mock things like the set of changed files).

We may eventually have some code we want to share between `eng/tools` and `.github`, and technically this is no problem as long as we write it in pure JS.